src/Classes/Helicoptere.php

```diff
@@ -0,0 +1,135 @@
+                }else{
+                    $this->$y+=$distance;
+                }
+            break;
+            //si vers la droite avance en x
+            case 90:
+                if(($this->$x+$distance)>=20){
+                    $this->$x=20;
+                }else{
+                    $this->$x+=$distance;
+                }
+            break;
+            //si vers le bas recule en y
+            case 180:
+                if(($this->$y-$distance)<=0){
+                    $this->$y=0;
+                }else{
+                    $this->$y-=$distance;
+                }
+            break;
+            //si versla gauche recule en x
+            case 270:
+                if(($this->$x-$distance)<=0){
+                    $this->$x=0;
+                }else{
+                    $this->$x-=$distance;
+                }
+            break;
+            //si diagonale haut droite avance y et x
+            case 45:
+                //il faut vérifier si on peut avancer àchaque fois
+                $resDistance=$distance;
+                while($distance>0){
+                    if(($this->$x<20)&&($this->$y<12)){
+                        $this->$x+=1;
+                        $this->$y+=1;
+                    }
+                    $distance-=1;
+                }
+            break;
+            //si diagonale bas droite recule y et avance x
+            case 135:
+                //il faut vérifier si on peut avancer àchaque fois
+                $resDistance=$distance;
+                while($distance>0){
+                    if(($this->$x<20)&&($this->$y>0)){
+                        $this->$x+=1;
+                        $this->$y-=1;
+                    }
+                    $distance-=1;
+                }
+            break;
+            //si diagonale bas gauche recule y et x
+            case 225:
+                //il faut vérifier si on peut avancer àchaque fois
+                $resDistance=$distance;
+                while($distance>0){
+                    if(($this->$x>0)&&($this->$y>0)){
+                        $this->$x-=1;
+                        $this->$y-=1;
+                    }
+                    $distance-=1;
+                }
+            break;
+            //si diagonale haut gauche avance y et recule x
+            case 315:
+                //il faut vérifier si on peut avancer àchaque fois
+                $resDistance=$distance;
+                while($distance>0){
+                    if(($this->$x>0)&&($this->$y<12)){
+                        $this->$x-=1;
+                        $this->$y+=1;
+                    }
+                    $distance-=1;
+                }
+            break;
+            default:
+            //tant pis
+            printf("Erreur, angle non valable (très content)");
+            break;
+        }
+    }
+
+    function seReorienter($angle){
+        $this->$direction+=$angle;
+    }
+
+    function prendreDegats(){
+        if($this->$z > 0) {
```

Comment:
You should rather test conditions before the operations, you're less prone to forgetting testing them.

By example:
```
if($this->z <= 0){
    throw
 new Exception("You are already crashed on the floor");
}
$this->$z -= 1;
```